src/main/java/com/example/demo/services/LeaseManagementDropBoxImpl.java

```diff
@@ -126,13 +127,16 @@ public LeaseDTO getLeaseStatus(Long leaseId) throws ApiException, EmptyResultDat
             lease.setStatus(DocStatus.CANCELED);
         }
         }
-        List<Tenant> signers = lease.getTenants();
-
+        List<TenantDto> signers = new ArrayList<>();
+        for(Tenant tenant: lease.getTenants()){
+            signers.add(TenantDto.builder().username(tenant.getUser().getUsername()).name(tenant.getUser().getName()).email(tenant.getUser().getEmail()).build());
+        }
```

Comment:
This looks quite complex. If all the entities have a mapper. You could just map the lease response that you get back from the ` save()`  to an DTO with the mapper.